openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-scriptrunner/src/tools/ScriptRunner/ScriptRunner.vue

```diff
@@ -1783,9 +1783,14 @@ end
     // Called by the FileOpenDialog to set the file contents
     setFile({ file, locked, breakpoints }, local = false) {
       this.files = {} // Clear the cached file list
+      // Split off the ' *' which indicates a file is modified on the server
+      let newFilename = file.name.split('*')[0]
       if (local === false) {
-        this.unlockFile() // first unlock what was just being edited
-        this.lockedBy = locked
+        // We only need to unlock if the file is different
+        if (this.filename !== newFilename) {
+          this.unlockFile() // first unlock what was just being edited
+          this.lockedBy = locked
+        }
```

Comment:
This is actually the main thing I wanted to fix. This prevents the read-only lock warning when you reload the existing file.